include/up-cpp/datamodel/builder/UMessage.h

```diff
@@ -227,8 +228,24 @@ struct UMessageBuilder {
 	/// @return A built message with the provided payload data embedded.
 	[[nodiscard]] v1::UMessage build(builder::Payload&&) const;
 
+	/// @brief Access the attributes of the message being built.
+	/// @return A reference to the attributes of the message being built.
+	[[nodiscard]] const v1::UAttributes& attributes() const { return attributes_; }
+
 private:
-	UMessageBuilder();
+	/// @brief Constructs a UMessageBuilder with the provided attributes.
+	///
+	/// @param msgType
+	/// @param source
+	/// @param sink
+	/// @param request_id
+	UMessageBuilder(v1::UMessageType msgType,
+					v1::UUri&& source,
+					v1::UUri&& sink={},
+					v1::UUID&& request_id={});
```

Comment:
Use `std::optional` to clearly convey whether a parameter was set or not. Default construction of message objects may not be the right behavior and is hard to check.

Reply:
Done, however I still have to provide default std::nullopt to it. So essentially, am still doing similar thing only using a different style..